File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/view/ToolView.java
package com.savvasdalkitsis.gameframe.draw.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

import com.savvasdalkitsis.gameframe.draw.model.DrawingTool;
import com.savvasdalkitsis.gameframe.draw.model.Tools;

public class ToolView extends ImageView {

    private DrawingTool drawingTool;

    public ToolView(Context context) {
        super(context);
    }

    public ToolView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ToolView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    protected DrawingTool getDrawingTool() {
        return drawingTool;
    }

    public void bind(Tools tool) {
        this.drawingTool = tool.getTool();
        setImageResource(tool.getIcon());
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/draw/view/ToolView.kt
package com.savvasdalkitsis.gameframe.feature.draw.view

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView

import com.savvasdalkitsis.gameframe.feature.draw.model.DrawingTool
import com.savvasdalkitsis.gameframe.feature.draw.model.Tools

class ToolView : ImageView {

    var drawingTool: DrawingTool? = null
        private set

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun bind(tool: Tools) {
        this.drawingTool = tool.tool
        setImageResource(tool.icon)
    }
}
